src/CVRF/cvrf_priv.c

```diff
@@ -0,0 +1,1518 @@
+// DocumentPublisher
+#define TAG_PUBLISHER BAD_CAST "DocumentPublisher"
+#define TAG_PUBLISHER_CONTACT BAD_CAST "ContactDetails"
+#define TAG_PUBLISHER_ISSUING_AUTHORITY BAD_CAST "IssuingAuthority"
+//Document
+#define TAG_DOCUMENT_TRACKING BAD_CAST "DocumentTracking"
+#define TAG_TRACKING_IDENTIFICATION BAD_CAST "Identification"
+#define TAG_TRACKING_ID BAD_CAST "ID"
+#define TAG_TRACKING_ALIAS BAD_CAST "Alias"
+#define TAG_TRACKING_STATUS BAD_CAST "Status"
+#define TAG_TRACKING_VERSION BAD_CAST "Version"
+#define TAG_TRACKING_INIT_RELEASE BAD_CAST "InitialReleaseDate"
+#define TAG_TRACKING_CUR_RELEASE BAD_CAST "CurrentReleaseDate"
+#define TAG_GENERATOR BAD_CAST "Generator"
+#define TAG_GENERATOR_ENGINE BAD_CAST "Engine"
+#define TAG_GENERATOR_DATE BAD_CAST "Date"
+// Product Tree
+#define TAG_CVRF_PRODUCT_TREE BAD_CAST "ProductTree"
+#define TAG_BRANCH BAD_CAST "Branch"
+#define ATTR_BRANCH_NAME BAD_CAST "Name"
+#define TAG_CVRF_PRODUCT_NAME BAD_CAST "FullProductName"
+#define ATTR_PRODUCT_ID BAD_CAST "ProductID"
+//Relationship
+#define TAG_RELATIONSHIP BAD_CAST "Relationship"
+#define ATTR_PRODUCT_REFERENCE BAD_CAST "ProductReference"
+#define ATTR_RELATES_TO_REF BAD_CAST "RelatesToProductReference"
+#define TAG_GROUP_ID BAD_CAST "GroupID"
+// Vulnerabilities
+#define TAG_VULNERABILITY BAD_CAST "Vulnerability"
+#define ATTR_VULNERABILITY_ORDINAL BAD_CAST "Ordinal"
+#define TAG_VULNERABILITY_TITLE BAD_CAST "Title"
+#define TAG_VULNERABILITY_ID BAD_CAST "ID"
+#define TAG_VULNERABILITY_DISCOVERY BAD_CAST "DiscoveryDate"
+#define TAG_VULNERABILITY_RELEASE BAD_CAST "ReleaseDate"
+#define TAG_VULNERABILITY_CVE BAD_CAST "CVE"
+#define TAG_VULNERABILITY_CWE BAD_CAST "CWE"
+#define TAG_CVRF_PRODUCT_STATUSES BAD_CAST "ProductStatuses"
+#define TAG_CVRF_PRODUCT_STATUS BAD_CAST "Status"
+#define TAG_PRODUCT_ID BAD_CAST "ProductID"
+#define ATTR_STATUS_TYPE BAD_CAST "Type"
+// Remediations
+#define TAG_REMEDIATIONS BAD_CAST "Remediations"
+#define TAG_REMEDIATION BAD_CAST "Remediation"
+#define TAG_REMEDIATION_DESC BAD_CAST "Description"
+#define TAG_REMEDIATION_URL BAD_CAST "URL"
+// namespaces
+#define CVRF_NS BAD_CAST "http://www.icasi.org/CVRF/schema/cvrf/1.1"
+#define PROD_NS BAD_CAST "http://www.icasi.org/CVRF/schema/prod/1.1"
+#define VULN_NS BAD_CAST "http://www.icasi.org/CVRF/schema/vuln/1.1"
+
+/* End of XML variable definitions
+ *
+ ***************************************************************************/
+
+/****************************************************************************
+ * Parsing functions
+ *
+ */
+
+struct cvrf_index *cvrf_index_parse_xml(struct oscap_source *index_source) {
+
+	__attribute__nonnull__(index_source);
+
+	struct cvrf_index *index = cvrf_index_new();
+	cvrf_index_set_index_file(index, oscap_source_readable_origin(index_source));
+	struct oscap_stringlist *file_list = oscap_stringlist_new();
+	struct cvrf_model *model;
+
+	FILE* file = fopen(oscap_source_get_filepath(index_source), "r");
```

Comment:
this will only work for oscap_source opened from local files. Use the oscap_source API to read and parse, don't use this clutch that converts it to a file.